src/core/Merger.h

```diff
@@ -33,7 +33,7 @@ class Merger : public QObject
     Merger(const Group* sourceGroup, Group* targetGroup);
     void setForcedMergeMode(Group::MergeMode mode);
     void resetForcedMergeMode();
-    bool merge();
+    QStringList merge();
```

Comment:
Perhaps now is a good time to introduce the option to "dry run" the merge function. We want to replicate this behavior on the GUI side as well. Can you make this: ```QStringList merge(bool dryrun = false);```

You'll have to propagate that down to the individual merge functions as well. Unless you think it would be better to introduce a ```m_dryrun``` class variable and a function such as ```setDryrun(bool)```.

Reply:
This is how I intended to implement the dry run at first, with a new `dryRun` argument to the `merge` function. I feel like it's much easier to perform the changes anyway in `merge`, but only commit the changes if we're not performing a dry run. The GUI could do the same thing by merging to a temporary copy of the database if the dry run option is set. Otherwise we have to trickle down the boolean param and verify it everywhere we perform changes. @droidmonkey what do you think?

Reply:
Ah I see, so merge in memory only, show the change list, if accepted replace the database with the in memory copy. That is a good idea (also saves a repeat merge operation).